packages/ui/src/components/Button/index.tsx

```diff
@@ -1,41 +1,104 @@
+import { I18nKey } from '@logto/phrases-ui';
 import classNames from 'classnames';
-import { ReactNode } from 'react';
+import { HTMLProps, ReactElement, ReactNode, useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 
+import DangerousRaw from '../../../../console/src/components/DangerousRaw';
+import { Ring as Spinner } from '../../../../console/src/components/Spinner';
 import * as styles from './index.module.scss';
 
-export type Props = {
+export type ButtonType =
+  | 'primary'
+  | 'secondary'
+  | 'danger'
+  | 'outline'
+  | 'plain'
+  | 'default'
+  | 'branding';
+
+type BaseProps = Omit<HTMLProps<HTMLButtonElement>, 'type' | 'size' | 'title'> & {
   htmlType?: 'button' | 'submit' | 'reset';
+  type?: ButtonType;
+  size?: 'small' | 'medium' | 'large';
   isDisabled?: boolean;
-  className?: string;
-  children: ReactNode | Record<string, unknown>;
-  type?: 'primary' | 'secondary' | 'outline';
-  size?: 'small' | 'large';
-  onClick?: React.MouseEventHandler;
+  isLoading?: boolean;
+  loadingDelay?: number;
+};
+
+type TitleButtonProps = BaseProps & {
+  title: I18nKey | ReactElement<typeof DangerousRaw>;
+  icon?: ReactNode;
+  properties?: Record<string, string>;
+};
+
+type IconButtonProps = BaseProps & {
+  title?: I18nKey | ReactElement<typeof DangerousRaw>;
+  icon: ReactNode;
+  properties?: Record<string, string>;
 };
 
+export type Props = TitleButtonProps | IconButtonProps;
+
 const Button = ({
   htmlType = 'button',
-  type = 'primary',
-  size = 'large',
-  isDisabled,
+  type = 'default',
+  size = 'medium',
```

Comment:
The update of the default value here might affect lots of button references here.